### ISSUE TITLE
Issue with constructor generation in 1.16.22

### ISSUE THREAD

**Issue by davidgoate** [rzwitserloot/lombok#1731]
Today I tried to update from spring boot `1.5.13.RELEASE` to `1.5.14.RELEASE` which [ships a new version of lombok](https://github.com/spring-projects/spring-boot/commit/884bd9008c5af60b5e2bd717042c99fcc80fa3ce)  - version `1.16.20` -> `1.16.22`.

Now my project (which used to build fine) fails with many complication errors. 

E.g.:

>[ERROR] /PersistentCommission.java:[13,1] constructor PersistentCommission() is already defined in class PersistentCommission.

Source is: 

```java
@Data
@NoArgsConstructor
public class PersistentCommission implements PersistentAmountWithPercentage {

    private PersistentAmount amount;
    private BigDecimal percentage;

    public PersistentCommission(final PriceProto.AmountWithPercentage commissionDto) {
        this.amount = new PersistentAmount(commissionDto.getAmount());
        this.percentage = new BigDecimal(commissionDto.getPercentage());
    }

}
```

Is this expected behaviour and actually an error in my code/use of lombok or could this be a bug? It certainly seems like a non-backwards compatible change. I'm happy to change my code, but first I'd like to understand why.

**Comment by Maaartinus** [comment]
This is probably related to this [change](https://projectlombok.org/changelog):

> FEATURE: Private no-args constructor for @Data and @Value to enable deserialization frameworks (like Jackson) to operate out-of-the-box. Use lombok.noArgsConstructor.extraPrivate = false to disable this behavior.

This was no good idea and was fixed in 1.18. Moreover, it looks like there was a bug leading to generating 
`@NoArgsConstructor` twice.

Either install 1.18 or follow the above advice.

**Comment by davidgoate** [comment]
@Maaartinus thanks, upgrading fixed it.
